src/components/MultiHopTrade/MultiHopTrade.tsx

```diff
@@ -76,25 +81,38 @@ const TradeRoutes = memo(({ isCompact }: TradeRoutesProps) => {
 
   const tradeInputRef = useRef<HTMLDivElement | null>(null)
 
+  const shouldUseTradeQuotes = useMemo(() => {
+    // We only want to fetch quotes when the user is on the trade input or quote list route
+    return [TradeRoutePaths.Input, TradeRoutePaths.QuoteList].includes(
+      location.pathname as TradeRoutePaths,
+    )
+  }, [location.pathname])
+
   return (
-    <AnimatePresence mode='wait' initial={false}>
-      <Switch location={location}>
-        <Route key={TradeRoutePaths.Input} path={TradeRoutePaths.Input}>
-          <TradeInput isCompact={isCompact} tradeInputRef={tradeInputRef} />
-        </Route>
-        <Route key={TradeRoutePaths.Confirm} path={TradeRoutePaths.Confirm}>
-          <MultiHopTradeConfirm />
-        </Route>
-        <Route key={TradeRoutePaths.VerifyAddresses} path={TradeRoutePaths.VerifyAddresses}>
-          <VerifyAddresses />
-        </Route>
-        <Route key={TradeRoutePaths.QuoteList} path={TradeRoutePaths.QuoteList}>
-          <QuoteListRoute
-            height={tradeInputRef.current?.offsetHeight ?? '500px'}
-            width={tradeInputRef.current?.offsetWidth ?? 'full'}
-          />
-        </Route>
-      </Switch>
-    </AnimatePresence>
+    <>
+      <AnimatePresence mode='wait' initial={false}>
+        <Switch location={location}>
+          <Route key={TradeRoutePaths.Input} path={TradeRoutePaths.Input}>
+            <TradeInput isCompact={isCompact} tradeInputRef={tradeInputRef} />
+          </Route>
+          <Route key={TradeRoutePaths.Confirm} path={TradeRoutePaths.Confirm}>
+            <MultiHopTradeConfirm />
+          </Route>
+          <Route key={TradeRoutePaths.VerifyAddresses} path={TradeRoutePaths.VerifyAddresses}>
+            <VerifyAddresses />
+          </Route>
+          <Route key={TradeRoutePaths.QuoteList} path={TradeRoutePaths.QuoteList}>
+            <QuoteListRoute
+              height={tradeInputRef.current?.offsetHeight ?? '500px'}
+              width={tradeInputRef.current?.offsetWidth ?? 'full'}
+            />
+          </Route>
+        </Switch>
+      </AnimatePresence>
+      {/* Stop polling for quotes by unmounting the hook. This prevents trade execution getting */}
+      {/* corrupted from state being mutated during trade execution. */}
+      {/* TODO: move the hook into a react-query or similar and pass a flag  */}
+      {shouldUseTradeQuotes ? <GetTradeQuotes /> : null}
```

Comment:
Yeah this feels like a bit of an anti-pattern, but understand wanting to use the min diff here to unrug the release pipeline and `develop`.